components/suggestion/suggestion_box.jsx

```diff
@@ -302,8 +310,11 @@ export default class SuggestionBox extends React.Component {
                 provider.handleCompleteWord(term, matchedPretext);
             }
         }
+
+        // override if user finished typing term before results returned from API
         if (shouldEmitWordSuggestion) {
-            GlobalActions.emitCompleteWordSuggestion(this.suggestionId);
+            const override = pretext.endsWith(term);
+            GlobalActions.emitCompleteWordSuggestion(this.suggestionId, '', override);
```

Comment:
If the user finished typing before the API returned, suggestions became outdated but would still appear in the `SuggestionBox` which was bad UX,

With an override flag, it avoids having the setCompletePending flag to true check `stores/suggestion_store.jsx` line: 319